util/pkg/vfs/azureclient.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"golang.org/x/oauth2"
+	"k8s.io/klog/v2"
+)
+
+const (
+	storageResourceID = "https://storage.azure.com/"
+)
+
+type azureClient struct {
+	p           pipeline.Pipeline
+	accountName string
+}
+
+func (c *azureClient) newContainerURL(containerName string) (*azblob.ContainerURL, error) {
+	u, err := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s", c.accountName, containerName))
+	if err != nil {
+		return nil, err
+	}
+	// Create a ContainerURL object that wraps the container URL and a request
+	// pipeline to make requests.
+	cURL := azblob.NewContainerURL(*u, c.p)
+	return &cURL, nil
+}
+
+func newAzureClient() (*azureClient, error) {
+	accountName := os.Getenv("AZURE_STORAGE_ACCOUNT")
```

Comment:
These environment variable names seem unusual to me but I confirmed the azure-cli uses them so it's best to be consistent :)

Reply:
Ah, yes.. :) Any suggestion appreciated here. I decided to use the env var as we need to pass the information to nodeup, protokube, and all other components that need to access a cluster config.